Hardit_Singh/bankAccount.rb

```diff
@@ -0,0 +1,59 @@
+class BankAccount
+	attr_reader	:accountNumber,
+				:userName,
+				:checkingBalance,
+				:savingBalance
+
+	def initialize(userName,balance=0)		
+		@accountNumber = genAccountNumber
+		@userName = userName
+		@checkingBalance = balance
+		@savingBalance = balance
+		@@noOfAccounts += 1
+		self
+	end
+
+	def get(req)
+		eval"@#{req}"
+	end
+
+	def deposit(amount,acctype)
+		(['checking','saving'].include? acctype) ?  (eval"@#{acctype}Balance+=#{amount}") : 'Not a valid account type'

```

Comment:
Ternaries are for simple operations or variable settings. That eval is jusy messy. A quick exit with a return checking for the include is better.